### PR TITLE
policy: More defensive code validating svc:DurationSeconds

### PR DESCRIPTION
**Pull request by vadmeste**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
This does not fix any current issue, but merging https://github.com/minio/madmin-go/pull/282 
can loose the validation of the service account expiration time and create a security issue.

Add more defensive code for now. In the future, we should avoid doing validation in another library.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/admin-handlers-users.go [hunk 1]
```diff
@@ -2453,11 +2457,16 @@ func (a adminAPIHandlers) ImportIAM(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func addExpirationToCondValues(exp *time.Time, condValues map[string][]string) {
+func addExpirationToCondValues(exp *time.Time, condValues map[string][]string) error {
 	if exp == nil {
-		return
+		return nil
+	}
+	dur := exp.Sub(time.Now())
```
Comment (harshavardhana) [inline cmd/admin-handlers-users.go:2464]:
```suggestion
	dur := exp.Sub(time.Now())
```

check if `exp.IsZero()` is true or `exp.Equal(time.Unix(0,0))`  return `nil`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/admin-handlers-users.go (hunk 1 above)
Comment (klauspost) [reply cmd/admin-handlers-users.go:2464]: Both would make the `if dur <= 0 {` below fail AFAICT, so it seems a bit superfluous.

**Review by klauspost — APPROVED** [review]
LGTM

File: cmd/admin-handlers-users.go (hunk 1 above)
Comment (vadmeste) [reply cmd/admin-handlers-users.go:2464]: @klauspost though Expiration is a pointer, some clients send a non nil expiration which is zero to mean I did not set any expiration in this request. I believe this is what Harsha meant. Fixed it

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**